pkg/odo/cli/logs/logs.go

```diff
@@ -116,11 +117,15 @@ func (o *LogsOptions) Run(ctx context.Context) error {
 		mode = odolabels.ComponentAnyMode
 	}
 
+	ns := ""
+	if o.clientset.KubernetesClient != nil {
```

Comment:
I just noticed that because the Kubernetes Client is nilable, we might need to check that we are using it in a safe manner.
And indeed, we have a panic when there is no Kubernetes client:
```
$ KUBECONFIG=/dev/null odo logs --platform cluster
============================================================================
⚠ Experimental mode enabled. Use at your own risk.
More details on https://odo.dev/docs/user-guides/advanced/experimental-mode
============================================================================

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x2a2681d]

goroutine 90 [running]:
github.com/redhat-developer/odo/pkg/logs.(*LogsClient).getPodsForSelector(0xc00047a3f0, {0xc000069290, 0x82}, {0x0, 0x0}, 0xc000138b40?)
        /home/asoro/work/projects/odo/pkg/logs/logs.go:123 +0xbd
github.com/redhat-developer/odo/pkg/logs.(*LogsClient).getLogsForMode(0xc00047a3f0, {0x355b748, 0xc000abfda0}, {0xc0001389c0?, 0xc000138a20?, 0xc000138a80?}, {0x0, 0x0}, {0xc000d02258, 0x17}, ...)
        /home/asoro/work/projects/odo/pkg/logs/logs.go:97 +0x246
created by github.com/redhat-developer/odo/pkg/logs.(*LogsClient).GetLogsForMode
        /home/asoro/work/projects/odo/pkg/logs/logs.go:56 +0x1d2
```

Not necessarily related to this PR, so I've created a separate issue (https://github.com/redhat-developer/odo/issues/6555) for this, as it also happens on the `main` branch.